File: apps/php/design-pattern/src/TemplateMethod/PlainTextReport.php
<?php

declare(strict_types=1);

namespace DesignPattern\TemplateMethod;

class PlainTextReport extends Report
{
    protected function outputHead(): array
    {
        return ["**** {$this->title} ****", ''];
    }

    protected function outputLine(string $line): array
    {
        return [$line];
    }
}

File: apps/java/design-pattern/src/main/java/pattern/templatemethod/PlainTextReport.java
package pattern.templatemethod;

/**
 * プレーンテキスト形式のレポート出力
 */
public class PlainTextReport extends Report {

    @Override
    protected void outputHead(StringBuilder sb) {
        sb.append("**** ").append(title).append(" ****\n");
        sb.append("\n");
    }

    @Override
    protected void outputLine(StringBuilder sb, String line) {
        sb.append(line).append("\n");
    }
}
